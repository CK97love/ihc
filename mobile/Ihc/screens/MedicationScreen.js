import React, { Component } from 'react';
import {
  StyleSheet,
  Button,
  CheckBox,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import * as data from '../services/FakeDataService';
import MedicationTable, {tableStyles} from '../components/MedicationTable';

export default class MedicationScreen extends Component<{}> {
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
      updates: [],
      error: null
    };
  }

  backToPatient = () => {
    this.props.navigator.push({
      screen: 'Ihc.PatientHomeScreen',
      title: this.props.patientInfo.name,
      passProps: { patientInfo: this.props.patientInfo }
    });
  }

  // TODO when soap screen is made
  goToSoap = () => {
    this.props.navigator.push({
      screen: 'Ihc.PatientSelectScreen',
      title: 'Select patient'
    });
  }

  // TODO buttons
  refillMedication = (prevDrugUpdate) => {
  }

  changeMedication = (prevDrugUpdate) => {
  }

  discontinueMedication = (prevDrugUpdate) => {
  }

  createNewMedication = () => {
  }

  render() {
    if(this.props.loading) {
      return (
        <View style={styles.container}>
          <Text style={styles.title}>
            {this.props.patientInfo.name}'s Medications
          </Text>

          <Text>Loading</Text>

          <Button onPress={this.backToPatient}
            title="Back to patient" />
          <Button onPress={this.goToSoap}
            title="To SOAP" />
        </View>
      );
    }

    return (
      <View style={styles.container}>
        <Text style={styles.title}>
          {this.props.patientInfo.name}'s Medications
        </Text>

        <MedicationTable style={styles.table} />

        <Button onPress={this.backToPatient}
          title="Back to patient" />
        <Button onPress={this.goToSoap}
          title="To SOAP" />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  table: {
  }
});
