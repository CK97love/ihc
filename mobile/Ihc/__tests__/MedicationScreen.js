import 'react-native';
import React from 'react';
import MedicationScreen from '../screens/MedicationScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const fakeNavigator = { setOnNavigatorEvent: (param) => {} };
  const json = renderer.create(
    <MedicationScreen navigator={fakeNavigator} />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
