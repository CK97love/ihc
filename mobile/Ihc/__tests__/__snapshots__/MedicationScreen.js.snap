// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          's Medications
        </Text>
        <Text>
          R: Refill, D: Change Dose, X: Cancel
        </Text>
        <Text>
          T: Taking, N: Not Taking, I: Taking incorrectly
        </Text>
      </View>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flex": 0,
          }
        }
        horizontal={true}
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              No data to show
            </Text>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
            "margin": 4,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Save checkmarks
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            New Medication
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Completed Prescribing
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Filled
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
